eclair-core/src/main/scala/fr/acinq/eclair/channel/Commitments.scala

```diff
@@ -144,9 +144,17 @@ case class Commitments(channelVersion: ChannelVersion,
     localCommit.spec.htlcs.collect(incoming).filter(nearlyExpired)
   }
 
-  def addLocalProposal(proposal: UpdateMessage): Commitments = Commitments.addLocalProposal(this, proposal)
-
-  def addRemoteProposal(proposal: UpdateMessage): Commitments = Commitments.addRemoteProposal(this, proposal)
+  /**
+   * Return a fully signed commit tx, that can be published as-is.
+   */
+  def fullySignedLocalCommitTx(keyManager: ChannelKeyManager): CommitTx = {
```

Comment:
Why not name this `signLocalCommitTx`?

Reply:
I could, but then it would make sense to provide the commit tx in the argument (and then we would have to make sure it is consistent with the commitment object itself. I also flip-flopped with moving this to `Helpers`. We're kind of in-between.

Reply:
You're right, as a method of the `Commitments` object your naming makes more sense.
Let's keep it like that, and I think it makes sense inside the `Commitments` object (at least for now).